### PR TITLE
feat: Custom Task Turn to Main Line: dde-file-manager Clears Search History

### PR DESCRIPTION
**Pull request by liyigang1**
Custom Task Turn to Main Line: dde-file-manager Clears Search History

Log: Custom Task Turn to Main Line: dde-file-manager Clears Search History
Task: https://pms.uniontech.com/task-view-341677.html

### CODE REVIEW

**Comment by Johnson-zs** [comment]
PMS链接错了？

**Comment by liyigang1** [comment]
> PMS链接错了？

已修改

File: src/dfm-base/base/application/application.h [hunk 1]
```diff
@@ -60,6 +60,7 @@ class Application : public QObject
         kShowDeleteConfirmDialog,   // display the delete confirmation dialog
         kHideLoopPartitions,   // hide loop partitions
         kShowThunmbnailInRemote,   // show file thumbnail in remote dir
+        KDisplaySearchHistory, // display search history
```
Comment (Johnson-zs) [inline src/dfm-base/base/application/application.h]: 配置使用DConfig，不得再使用json

**Review by Johnson-zs — CHANGES_REQUESTED** [review]

File: src/dfm-base/base/application/application.h (hunk 1 above)
Comment (liyigang1) [reply src/dfm-base/base/application/application.h]: 已修改

**Review by Johnson-zs — APPROVED** [review]

**Pull request merged**